user/module/probe_openssl_lib.go

```diff
@@ -119,67 +118,63 @@ func (m *MOpenSSLProbe) detectOpenssl(soPath string) error {
 		return nil
 	}
 
-	sectionSize := int64(s.Offset)
+	sectionOffset := int64(s.Offset)
+	sectionSize := s.Size
+
+	r.Close()
 
 	_, err = f.Seek(0, 0)
 	if err != nil {
 		return err
 	}
 
-	ret, err := f.Seek(sectionSize, 0)
-	if ret != sectionSize || err != nil {
+	ret, err := f.Seek(sectionOffset, 0)
+	if ret != sectionOffset || err != nil {
 		return err
 	}
 
-	buf := make([]byte, s.Size)
-	if buf == nil {
-		return nil
-	}
+	versionKey := ""
 
-	_, err = f.Read(buf)
+	// e.g : OpenSSL 1.1.1j  16 Feb 2021
+	rex, err := regexp.Compile(`(OpenSSL\s\d\.\d\.[0-9a-z]+)`)
 	if err != nil {
-		return err
-	}
-
-	// 按照\x00 拆分  buf
-	var slice [][]byte
-	if slice = bytes.Split(buf, []byte("\x00")); slice == nil {
 		return nil
 	}
 
-	dumpStrings := make(map[uint64][]byte, len(slice))
-	length := uint64(len(slice))
+	buf := make([]byte, 1024*1024) // 1Mb
+	totalReadCount := 0
+	for totalReadCount < int(sectionSize) {
+		readCount, err := f.Read(buf)
 
-	var offset uint64
+		if err != nil {
+			m.logger.Printf("%s\t[f.Read] Error:%v\t", m.Name(), err)
+			break
+		}
 
-	for i := uint64(0); i < length; i++ {
-		if len(slice[i]) == 0 {
-			continue
+		if readCount == 0 {
+			break
 		}
 
-		dumpStrings[offset] = slice[i]
+		match := rex.Find(buf)
+		if match != nil {
+			versionKey = string(match)
+			break
+		}
 
-		offset += (uint64(len(slice[i])) + 1)
-	}
+		totalReadCount += readCount
```

Comment:
Can we make some retrospective adjustments here? For example, `totalReadCount += readCount-OpenSslVersionLen`? 

Move forward by an offset of length `OpenSslVersionLen` to address the previously mentioned issue? 

Additionally, `const OpenSslVersionLen = 30 // openssl version string length`.

Reply:
Nice observation @cfc4n , using this adjustment we would be able to cover the edge case. :tada:  Thanks !